src/Uno.UI/UI/Xaml/Input/FocusManager.wasm.cs

```diff
@@ -44,10 +46,30 @@ internal static void ProcessElementFocused(UIElement element)
 				_log.Value.LogDebug($"{nameof(ProcessElementFocused)}() focusedElement={GetFocusedElement()}, element={element}, searching for focusable parent control");
 			}
 
-			foreach (var parent in element.GetParents())
+			foreach (var candidate in element.GetParents().Prepend(element))
```

Comment:
Consider adding an optional `bool includeSelf` parameter to GetParents that defaults to false.

Reply:
Actually, there is already `GetAllParents` that will do that job